### PR TITLE
implement standar error

### PR DESCRIPTION
**Pull request by candrasaputra**

### CODE REVIEW

File: language/en-GB/en-GB.plg_vmpayment_xendit.ini [hunk 1]
```diff
@@ -12,7 +12,7 @@ VMPAYMENT_XENDIT_CONF_PUBLIC_KEY_TIP="Unique Public API Key given by Xendit. Cas
 VMPAYMENT_XENDIT_CONF_SECRET_KEY_TIP="Unique Secret API Key given by Xendit. Case Sensitive!"
 VMPAYMENT_XENDIT_CONF_PAYMENT_TYPE_LABEL="Payment Type"
 
-VMPAYMENT_XENDIT_PARAMETER_REQUIRED="Xendit API Keys have not been configured."
-VMPAYMENT_XENDIT_UNSUPPORTED_CURRENCY="Xendit Payment Gateway does not support currency other than IDR (Indonesian Rupiah)."
+VMPAYMENT_XENDIT_PARAMETER_REQUIRED="Please include your API key as an Authorization header. Code: 100009"
+VMPAYMENT_XENDIT_UNSUPPORTED_CURRENCY="%s is currently not supported. Code: 100008"
 VMPAYMENT_XENDIT_INVALID_AMOUNT="Xendit Payment Gateway does not support decimal order amount."
```
Comment (IreneGohtami) [inline language/en-GB/en-GB.plg_vmpayment_xendit.ini:17]: shopify also has this decimal amount error & I've asked yano to add a new generic error code for this in the sheet, later on you can check & see.

File: xendit.php [hunk 2]
```diff
@@ -1246,40 +1237,44 @@ static function getXenditErrorMessage ($response) {
...
-					'title' => 'CARD_DECLINED',
-					'message' => 'The bank that issued this card declined the payment but didn\'t tell us why. Try another card, or try calling your bank to ask why the card was declined.'
+					'title' => $failure_reason,
+					'message' => 'Card declined by the issuer bank. Please try with another card or contact the bank directly. code: 200013'
 				);
 			case 'INSUFFICIENT_BALANCE':
 				return array(
 					'title' => $failure_reason,
-					'message' => 'Your bank declined this payment due to insufficient balance. Ensure that sufficient balance is available, or try another card'
+					'message' => 'Card declined by the issuer bank. Please try with another card or contact the bank directly. Code: 200012'
```
Comment (IreneGohtami) [inline xendit.php:1252]: please refer to the excel sheet for `INSUFFICIENT_BALANCE` message.

File: xendit.php [hunk 3]
```diff
@@ -1246,40 +1237,44 @@ static function getXenditErrorMessage ($response) {
 	static function getXenditFailureMessage ($failure_reason) {
         switch ($failure_reason) {
 			case 'CARD_DECLINED':
+				return array(
+					'title' => $failure_reason,
```
Comment (IreneGohtami) [inline xendit.php:1241]: If the title is no longer used, maybe just removed it altogether?

File: xendit.php [hunk 4]
```diff
@@ -1223,69 +1223,37 @@ static function getNotificationUrl ($order) {
...
-					'title' => $failure_reason,
-					'message' => 'Authentication process failed. Please try again'
-				);
+				return 'The authentication process failed. Please try again. Code: 200001';
             default:
-				return array(
-					'title' => $failure_reason,
-					'message' => 'Failed to process transaction.'
-				);
+				return 'Failed to process transaction.';
```
Comment (IreneGohtami) [inline xendit.php:1256]: Wanna use this default error message instead? 
`We encountered an issue while processing the checkout. Please try again. Code: 100007' `

File: xendit/tmpl/cc_payment.php [hunk 5]
```diff
@@ -108,7 +108,11 @@ function xenditClickCC() {
 						
 						Xendit.card.createToken(data, function(tokenErr, tokenResponse) { // on tokenization response
 							if (tokenErr) { // how to display VM error style in here?
-								alert(tokenErr.error_code + ": " + tokenErr.message);
+								if(tokenErr.error_code === 'VALIDATION_ERROR') {
+									alert('Inputted credit card data (can be CC number, expiry date, CVN) is invalid. Code: 200003');
```
Comment (IreneGohtami) [inline xendit/tmpl/cc_payment.php:112]: `Please verify that the credit card information is correct. Code: 200003`

**Review by yanoandri — APPROVED** [review]

**Pull request merged**